binaries/cuprated/src/rpc/json.rs

```diff
@@ -83,212 +125,1126 @@ pub(super) async fn map_request(
-    todo!()
+    let height = usize_to_u64(
+        blockchain_context::context(&mut state.blockchain_context)
+            .await?
+            .unchecked_blockchain_context()
+            .chain_height,
+    );
+
+    let target_height = blockchain_manager::target_height(&mut state.blockchain_manager).await?;
+
+    let peers = address_book::connection_info::<ClearNet>(&mut DummyAddressBook)
+        .await?
+        .into_iter()
+        .map(|info| SyncInfoPeer { info })
+        .collect();
+
+    let next_needed_pruning_seed =
+        blockchain_manager::next_needed_pruning_seed(&mut state.blockchain_manager)
+            .await?
+            .compress();
+
+    let spans = blockchain_manager::spans::<ClearNet>(&mut state.blockchain_manager).await?;
+
+    // <https://github.com/Cuprate/cuprate/pull/320#discussion_r1811063772>
+    let overview = String::from(FIELD_NOT_SUPPORTED);
+
+    Ok(SyncInfoResponse {
+        base: AccessResponseBase::OK,
+        height,
+        next_needed_pruning_seed,
+        overview,
+        peers,
+        spans,
+        target_height,
+    })
 }
 
+/// <https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/core_rpc_server.cpp#L3332-L3350>
 async fn get_transaction_pool_backlog(
-    state: CupratedRpcHandler,
-    request: GetTransactionPoolBacklogRequest,
+    mut state: CupratedRpcHandler,
+    _: GetTransactionPoolBacklogRequest,
 ) -> Result<GetTransactionPoolBacklogResponse, Error> {
-    todo!()
+    let backlog = txpool::backlog(&mut state.txpool_read)
+        .await?
+        .into_iter()
+        .map(|entry| TxBacklogEntry {
+            weight: entry.weight,
+            fee: entry.fee,
+            time_in_pool: entry.time_in_pool.as_secs(),
+        })
+        .collect();
+
+    Ok(GetTransactionPoolBacklogResponse {
+        base: ResponseBase::OK,
+        backlog,
+    })
 }
 
+/// <https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/core_rpc_server.cpp#L3352-L3398>
+async fn get_output_distribution(
+    mut state: CupratedRpcHandler,
+    request: GetOutputDistributionRequest,
+) -> Result<GetOutputDistributionResponse, Error> {
+    if state.is_restricted() && request.amounts != [1, 0] {
+        return Err(anyhow!(
+            "Restricted RPC can only get output distribution for RCT outputs. Use your own node."
+        ));
+    }
+
+    // 0 is placeholder for the whole chain
+    let req_to_height = if request.to_height == 0 {
+        helper::top_height(&mut state).await?.0.saturating_sub(1)
+    } else {
+        request.to_height
+    };
+
+    let distributions = request.amounts.into_iter().map(|amount| {
+        fn get_output_distribution() -> Result<Distribution, Error> {
+            todo!("https://github.com/monero-project/monero/blob/893916ad091a92e765ce3241b94e706ad012b62a/src/rpc/rpc_handler.cpp#L29");
+            Err(anyhow!("Failed to get output distribution"))
+        }
```

Comment:
This seems to be used by ZMQ too, I think it should be defined outside of `cuprated`. It may be changed by FCMP++ as well so I think an impl could wait.